macaroonbakery/tests/test_agent.py

```diff
@@ -139,12 +142,15 @@ def discharge(url, request):
 
         @urlmatch(path='.*/login')
         def login(url, request):
+            qs = parse_qs(urlparse(request.url).query)
```

Comment:
could we also assert that the method is GET not POST? (not that that's changed, but ISTM that it's worth checking).